src/battle_script_commands.c

```diff
@@ -9341,7 +9341,7 @@ static void Cmd_weightdamagecalculation(void)
     s32 i;
     for (i = 0; sWeightToDamageTable[i] != 0xFFFF; i += 2)
     {
-        if (sWeightToDamageTable[i] > GetPokedexHeightWeight(SpeciesToNationalPokedexNum(gBattleMons[gBattlerTarget].species), 1))
+        if (sWeightToDamageTable[i] > GetSpeciesHeightWeight(gBattleMons[gBattlerTarget].species, 1))
```

Comment:
I think that instead of creating a new function for this, we should modify `GetPokedexHeightWeight` instead, as modifying `battle_script_commands.c` can potentially create a conflict with BE.